build_tools/circle/build_doc.sh

```diff
@@ -169,14 +169,39 @@ affected_doc_paths() {
     fi
 }
 
+affected_doc_warnings() {
+    files=$(git diff --name-only origin/master...$CIRCLE_SHA1)
+    # Look for sphinx warnings only in files affected by the PR
+    if [ -n "$files" ]
+    then
+        for af in ${files[@]}
+        do
+          warn+=`grep WARNING ~/log.txt | grep $af`
+        done
+    fi
+    echo "$warn"
+}
+
 if [ -n "$CI_PULL_REQUEST" ]
 then
+    echo "The following documentation warnings may have been generated by PR #$CI_PULL_REQUEST:"
+    warnings=$(affected_doc_warnings)
+    if [ -z "$warnings" ]
+    then
+        warnings="/home/circleci/project/ no warnings"
+    fi
+    echo "$warnings"
+
     echo "The following documentation files may have been changed by PR #$CI_PULL_REQUEST:"
     affected=$(affected_doc_paths)
     echo "$affected"
     (
     echo '<html><body><ul>'
     echo "$affected" | sed 's|.*|<li><a href="&">&</a> [<a href="https://scikit-learn.org/dev/&">dev</a>, <a href="https://scikit-learn.org/stable/&">stable</a>]</li>|'
-    echo '</ul><p>General: <a href="index.html">Home</a> | <a href="modules/classes.html">API Reference</a> | <a href="auto_examples/index.html">Examples</a></p></body></html>'
+    echo '</ul><p>General: <a href="index.html">Home</a> | <a href="modules/classes.html">API Reference</a> | <a href="auto_examples/index.html">Examples</a></p>'
+    echo '<strong>Sphinx Warnings in affected files</strong><ul>'
+    echo "$warnings" | sed 's/\/home\/circleci\/project\//<li>/g'
```

Comment:
The end tag, `</li>`, is not placed.

Reply:
You are right. I couldn't find a way to parse the end of the warnings, and you may have more than one warning per file. Fortunately, it works... :) ... and it happens to conform to the HTML5 recommendations (see the case of [possible omission of end `li` tag](https://www.w3.org/TR/html52/dom.html#tag-omission-in-text-html)) while keeping the code quite simple.

Reply:
It seems to work on chrome, firefox, and safari, so I am okay with not including it.